source/Calamari.AzureAppService.Tests/AzureWebAppHealthCheckActionHandlerFixtures.cs

```diff
@@ -0,0 +1,173 @@
+﻿#nullable disable
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Calamari.Azure;
+using Calamari.Common.Plumbing.Proxies;
+using Calamari.Tests.Shared;
+using FluentAssertions;
+using Microsoft.Azure.Management.AppService.Fluent;
+using Microsoft.Azure.Management.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using NUnit.Framework;
+using OperatingSystem = Microsoft.Azure.Management.AppService.Fluent.OperatingSystem;
+
+namespace Calamari.AzureAppService.Tests
+{
+    [TestFixture]
+    class AzureWebAppHealthCheckActionHandlerFixtures
+    {
+        const string NonExistentProxyHostname = "non-existent-proxy.local";
+        const int NonExistentProxyPort = 3128;
+
+        IWebProxy? originalProxy;
+        string originalProxyHost;
+        string originalProxyPort;
+        
+        [Test]
+        public async Task WebAppIsFound_WithAndWithoutProxy()
+        {
+            IAzure azure = null;
+            IResourceGroup resourceGroup = null;
+            IWebApp webApp;
+            try
+            {
+                (azure, resourceGroup, webApp) = await SetUpAzureWebApp();
+                
+                await CommandTestBuilder.CreateAsync<HealthCheckCommand, Program>()
+                                        .WithArrange(context => SetUpVariables(context, resourceGroup.Name, webApp.Name))
+                                        .WithAssert(result => result.WasSuccessful.Should().BeTrue())
+                                        .Execute();
+                
+                // Here we verify whether the proxy is correctly picked up
+                // Since the proxy we use here is non-existent, health check to the same Web App should fail due this this proxy setting
+                SetLocalEnvironmentProxySettings(NonExistentProxyHostname, NonExistentProxyPort);
+                SetCiEnvironmentProxySettings(NonExistentProxyHostname, NonExistentProxyPort);
```

Comment:
Do we need to un-set these in the clean-up of the test? I'm wondering if we could end up with these settings impacting our `WebAppIsNotFound` now that we've moved them to be in the same fixture 😬

Reply:
Thanks Matt. Good catch. I've modified the set up and clean up a bit, so now it should work correctly in both tests:
- it keeps a record of the original settings before any test runs
- for the IsFound test, the proxy settings will be changed in test, and will be restored in the `TearDown` method after the test is finished ([even if the test terminates with an exception](https://docs.nunit.org/articles/nunit/writing-tests/attributes/teardown.html))
- for the IsNotFound test, the proxy settings will not change, but the `TearDown` method will run in the end anyway. This takes no effect since it just changes the settings to the original values (without the modification, this could change settings to `null` and `""` which may cause problems in other tests)